src/main/webapp/app/overview/course-exercises/course-exercises.component.html

```diff
@@ -1,73 +1,69 @@
-<div class="row">
-    <div class="col-12 col-lg-8">
-        <div class="mt-3 position-relative" *ngIf="course">
-            <div class="course-overview-controls" *ngIf="course!.exercises && course!.exercises.length > 0">
-                <div class="d-flex align-items-center me-2">
-                    <div class="me-2" aria-label="Filter Dropdown" ngbDropdown>
-                        <button
-                            class="btn"
-                            [ngClass]="{ 'btn-secondary': activeFilters.size === 0, 'btn-success': activeFilters.size > 0 }"
-                            ngbDropdownToggle
-                            id="filter-dropdown-button"
+<ng-template #controls>
+    <div class="course-overview-controls" *ngIf="course && course!.exercises && course!.exercises.length > 0">
```

Comment:
```suggestion
    <div class="course-overview-controls" *ngIf="course?.exercises?.length > 0">
```
Nitpick: For conciseness, optional chaining can be utilized but not necessarily required.

Reply:
I did actually try this. But TypeScript complained about it. I don't know why :D
![grafik](https://user-images.githubusercontent.com/23171488/146266567-34834585-3e87-4bd3-a59f-1fea63857c99.png)

Reply:
Oh I see, right. [Here](https://stackoverflow.com/questions/54884488/how-can-i-solve-the-error-ts2532-object-is-possibly-undefined#:~:text=The%20following%20will%20fail%20in%20strict%20TypeScript%2C%20as%20you%20are%20possibly%20comparing%20an%20undefined%20value%20with%20a%20number.) I found out it is warning against the possible risk of having `undefined > 0`. 
This seems to work but a little implicit: `*ngIf="course?.exercises?.length"`
All in all, it isn't a critical point and I already approved anyway😄